Adds some features on SSH/script plugin and fixing some issues (#85)
ovh/utask: pkg/plugins/builtin/ssh/README.md

This plugin connects to a remote system and performs a block of commands. It can extract variables from the shell back to the output of its enclosing step.

> Task will be considered successful if script returns exit code 0, otherwise, it will be considered as SERVER_ERROR. For some unrecoverable errors (for instance, invalid parameters), it is possible to add a configuration (`exit_codes_unrecoverable`) to consider some exit codes as CLIENT_ERROR.

Reviewer: ```suggestion
The step will be considered successful if the script returns exit code 0, otherwise, it will be considered as a `SERVER_ERROR` (and will be retried). For unrecoverable errors (for instance, invalid parameters), it is possible to configure a list of exit codes (see `exit_codes_unrecoverable`) that should halt the execution (`CLIENT_ERROR`).
```

| `target` | address of the remote machine
| `hops` | a list of intermediate addresses (bastions)
| `script` | multiline text, commands to be run on the machine's shell

> | `output_mode` | indicates how to retrieve output values ; valid values are: `auto-result` (default), `disabled`, `manual-delimiters`, `manual-lastline`

Reviewer: ```suggestion
| `output_mode` | indicates how to retrieve the output values ; valid values are: `auto-result` (default), `disabled`, `manual-delimiters`, `manual-lastline`
```

| `result` | an object to extract the values of variables from the machine's shell (only used when `output_mode` is configured to `auto-result`)
| `output_manual_delimiters` | array of 2 strings ; look for a JSON formatted string in the script output between specific delimiters (only used when `output_mode` is configured to `manual-delimiters`)
| `ssh_key` | private ssh key, preferrably retrieved from {{.config}}
| `ssh_key_passphrase` | passphrase for the key, if any

> | `exit_status_mapping` | an object to personalize the returned error class if exit_status is non-zero exit code; key should be an exit code (or an exit code range), value is a step state (CLIENT_ERROR or SERVER_ERROR)

Reviewer: ```suggestion
| `exit_codes_unrecoverable` | a list of non-zero exit codes (1, 2, 3, ...) or ranges (1-10, ...) which should be considered unrecoverable and halt execution ; these will be returned to the main engine as a `CLIENT_ERROR`
```

| `target` | address of the remote machine
| `hops` | a list of intermediate addresses (bastions)
| `script` | multiline text, commands to be run on the machine's shell
| `result` | an object to extract the values of variables from the machine's shell ; parameter conflicts with `output_disable_auto_result`.
| `output_disable_auto_result` | disables the bash TRAP function which formats an output line after script run (bool default `false`). `result` won't be usable.

> | `output_manual_delimiters` | personalize starting and ending pattern for uTask to look for a JSON-format output (array of 2 elements, default empty). `result` configuration will not be used.

Reviewer: ```suggestion
| `output_manual_delimiters` | look for a JSON formatted string in the script output between specific delimiters (array of 2 elements, default empty). `result` configuration will not be used.
```

    script: |-
      PID=$(systemctl show --property MainPID {{.input.serviceName}} | cut -d= -f2)
      SERVICE_UPTIME=$(ps -h -p ${PID} -o etimes)

>     output_mode: auto-result

Reviewer: ```suggestion
    # this is the default mode
    output_mode: auto-result
```

      PID=$(systemctl show --property MainPID {{.input.serviceName}} | cut -d= -f2)
      SERVICE_UPTIME=$(ps -h -p ${PID} -o etimes)
    # value extraction
    result:
      pid: $PID
      uptime: $SERVICE_UPTIME
    # credentials
    ssh_key: '{{.config.mySSHKey}}'
    output_disable_auto_result: false

>     # optional delimiters to look for an output -- not to use at the same time than result config

Reviewer: ```suggestion
    # optional delimiters to look for an output -- do not use with a result object configuration
```
